### ISSUE TITLE
[Feature] Make scrollIntoViewIfNeeded work for hidden elements

### ISSUE THREAD

**Issue by Nav-2d** [microsoft/playwright#13508]
### Discussed in https://github.com/microsoft/playwright/discussions/13454

<div type='discussions-op-text'>

<sup>Originally posted by **Nav-2d** April  9, 2022</sup>
Hello,

I had one scenario: I have an element that is `enabled and attached to DOM` but not `visible`. So , to click it I have to scroll to the element using:
await page.evaluate((loc) => document.querySelector(loc)?.scrollIntoView(), loc);
I can't use `scrollIntoViewIfNeeded()` either since it depends on element to be visible. I was wondering is my above solution the best or I can use something else? 

Thanks!</div>

Please see the above discussion for further details.

In summary, wondering if it would be nice to have `scrollIntoViewIfNeeded()` scroll to hidden elements. Please feel to close the request if it doesn't make sense. Thank you!

**Comment by aslushnikov** [comment]
@Nav-2d As per private discussion, here's what's going on:

- there's an element with 0 width and 0 height, but with some position in layout
- once the page is scrolled towards to the element, it's getting populated with some content

Running chromium's built-in `scrollintoViewIfNeeded` on this element would scroll to it and trigger code to populate the element. Makes sense for us to behave the same way! cc'ing @dgozman 

Here's a repro:

```html
<!-- demo.html -->
<div style="height: 2000px; text-align: center; border: 10px solid blue;">
  <h1>SCROLL DOWN</h1>
</div>
<span id=lazyload style="font-size:75px; background-color: green;"></span>
<script>
const lazyLoadElement = document.querySelector('#lazyload');
const observer = new IntersectionObserver((entries) => {
  if (entries.some(entry => entry.isIntersecting)) {
    lazyLoadElement.textContent = 'LAZY LOADED CONTENT';
    observer.disconnect();
  }
});
observer.observe(lazyLoadElement);
</script>
```

Script:

```js
// a.mjs
import { chromium } from '@playwright/test';

const browser = await chromium.launch({ headless: false });
const page = await browser.newPage();

await page.goto('file:///Users/andreylushnikov/tmp/demo.html');
// This should scroll to the element.
await page.locator('#lazyload').scrollIntoViewIfNeeded();
```
